### ISSUE TITLE
unnecessary-list-index-lookup inconsistent when assigning to iterating object in iterable unpacking

### ISSUE THREAD

**Issue by Tratori** [PyCQA/pylint#6788]
### Bug description
```python
# pylint: disable=missing-docstring
myList = [1, 2, 3]

for a, b in enumerate(myList):
    myList[a] = 2

for a, b in enumerate(myList):
    myList[a], _ = (2, 1)
```

### Command used

```shell
pylint test2.py
```

### Pylint output

```shell
************* Module test2
test2.py:8:4: R1736: Unnecessary list index lookup, use 'b' instead (unnecessary-list-index-lookup)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 8.00/10, +0.00)
```

### Expected behavior

I would expect the output for both loops to be the same. The latter loop leads to an unnecessary-list-index-lookup though.

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.9.12 (main, Jun  1 2022, 11:38:51) 
[GCC 7.5.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by jacobtylerwalls** [comment]
Thanks for the report, the second example looks like a false positive because the assumption of direct parentage here is violated (`subscript.parent` is a `<Tuple.tuple l.8 ...`):
https://github.com/PyCQA/pylint/blob/6e6927341aa38b2153a3ca7096f1a6447e135dbf/pylint/checkers/refactoring/refactoring_checker.py#L1921

cc/ @timmartin 

I also see we have similar code in three places 👀 (grep for "Early termination" to find them quickly).

**Comment by timmartin** [comment]
Thanks for the heads-up, I don't mind taking a look at this.
